### PR TITLE
Adds some tracks to the discoball for extreme awooing and a awooing sound affect

### PR DESCRIPTION
**Pull request by improvedname**
Awooooo!!

Tracks include as the following.
Awoo sound affect - https://www.youtube.com/watch?v=8AeQnH3JX_c
This is how it goes - https://www.youtube.com/watch?v=mjLbXwdqs18
Telephone number - https://www.youtube.com/watch?v=I0JVRcJLea8
Awakening wolf instinct - https://www.youtube.com/watch?v=V_uikyoC95c

:cl: Improvedname
tweak: adds more stuff to the discoball
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.) More variations then just doom and 2 versions of the shuffle duffle

### CODE REVIEW

**Review by deathride58 — CHANGES_REQUESTED** [review]
Good content, but you can modularize all of this. Only the awoo button in interact really needs to be touching the code.

File: modular_citadel/code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/obj/machinery/disco
+	/list/songs = list(
```
Comment (deathride58) [inline modular_citadel/code/game/machinery/dance_machine.dm]: You don't need /list/, you just need songs.

File: modular_citadel/code/game/machinery/dance_machine.dm [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+		new	/datum/track("This is how it goes", 						'sound/misc/Awoo1.ogg', 	2400, 	5),
+		new /datum/track("Telephone Number", 							'sound/misc/Awoo2.ogg', 	2400, 	6),
+		new /datum/track("Awoo", 										'sound/misc/Awoo3.ogg', 	2400, 	4),
+		new	/datum/track("Wolf instinct",								'sound/misc/Awoo4.ogg',		2400, 	20),
+		)
+
+		list/available = list()
```
Comment (deathride58) [inline modular_citadel/code/game/machinery/dance_machine.dm]: This is defined in a proc. You don't need this here.

File: modular_citadel/code/game/machinery/dance_machine.dm [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+		if ("awoo")
+			deejay('sound/misc/Awoo.ogg')
```
Comment (deathride58) [inline modular_citadel/code/game/machinery/dance_machine.dm]: This should go under topic. This is a proc thing.

File: code/game/machinery/dance_machine.dm [hunk 4]
```diff
@@ -176,7 +177,7 @@
 	if (QDELETED(src) || !active || charge < 5)
 		to_chat(usr, "<span class='warning'>The device is not able to play more DJ sounds at this time.</span>")
 		return
-	charge -= 5
+		charge -= 5
```
Comment (deathride58) [inline code/game/machinery/dance_machine.dm]: You don't need this whitespace change

File: code/game/machinery/dance_machine.dm [hunk 5]
```diff
@@ -16,10 +16,10 @@
...
 	var/list/sparkles = list()
 	var/static/list/songs = list(
-		new /datum/track("Engineering's Basic Beat", 					'sound/misc/disco.ogg', 	600, 	5),
-		new /datum/track("Engineering's Domination Dance", 				'sound/misc/e1m1.ogg', 		950, 	6),
-		new /datum/track("Engineering's Superiority Shimmy", 			'sound/misc/paradox.ogg', 	2400, 	4),
-		new /datum/track("Engineering's Ultimate High-Energy Hustle",	'sound/misc/boogie2.ogg',	1770, 	5),
+		new /datum/track("Basic Beat",				 					'sound/misc/disco.ogg', 	600, 	5),
+		new /datum/track("Domination Dance",			 				'sound/misc/e1m1.ogg', 		950, 	6),
+		new /datum/track("Superiority Shimmy", 							'sound/misc/paradox.ogg', 	2400, 	4),
+		new /datum/track("Ultimate High-Energy Hustle",					'sound/misc/boogie2.ogg',	1770, 	5),
```
Comment (deathride58) [inline code/game/machinery/dance_machine.dm]: These whitespace changes, reeeEE

Comment (deathride58) [reply code/game/machinery/dance_machine.dm]: Also you should copy and paste everything here into the overridden songs var in modular_citadel.

**Review by deathride58 — CHANGES_REQUESTED** [review]

File: modular_citadel/code/game/machinery/dance_machine.dm [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/obj/machinery/disco
+	/songs = list(
+		new	/datum/track("This is how it goes", 						'sound/misc/Awoo1.ogg', 	2400, 	5),
+		new /datum/track("Telephone Number", 							'sound/misc/Awoo2.ogg', 	2400, 	6),
+		new /datum/track("Awoo", 										'sound/misc/Awoo3.ogg', 	2400, 	4),
+		new	/datum/track("Wolf instinct",								'sound/misc/Awoo4.ogg',		2400, 	20),
+		)
+
+/obj/machinery/disco/Topic(href_list)
```
Comment (deathride58) [inline modular_citadel/code/game/machinery/dance_machine.dm:9]: Topic(href, href_list)

Also be sure to add a `. = ..()` and a `switch(href_list["action"])` here

**Review by deathride58 — CHANGES_REQUESTED** [review]

**Comment by improvedname** [comment]
Working on this later

**Pull request closed without merging**